onelogin_aws_cli/__init__.py

```diff
@@ -110,12 +115,64 @@ def get_saml_assertion(self):
                                headers, params)
             self.saml = res
 
-    def assume_role(self):
+    def get_arns(self):
         if not self.saml:
             self.get_saml_assertion()
+        # Parse the returned assertion and extract the authorized roles
+        awsroles = []
+        root = ET.fromstring(base64.b64decode(self.saml))
+
+        for saml2attribute in root.iter('{urn:oasis:names:tc:SAML:2.0:assertion}Attribute'):
+             if (saml2attribute.get('Name') == 'https://aws.amazon.com/SAML/Attributes/Role'):
+                 for saml2attributevalue in saml2attribute.iter('{urn:oasis:names:tc:SAML:2.0:assertion}AttributeValue'):
+                     awsroles.append(saml2attributevalue.text)
+
+        # Note the format of the attribute value should be role_arn,principal_arn
+        # but lots of blogs list it as principal_arn,role_arn so let's reverse
+        # them if needed
+        for awsrole in awsroles:
```

Comment:
faster, simpler with list comprehensions:

```python
aws_roles = [role.split(",") for role in aws_roles]
aws_roles = [(principal, role) for role, principal in aws_roles]
```

Also, easier to store them as a tuple